### ISSUE TITLE
App crashing on launch when running on real devices (framework linking)

### ISSUE THREAD

**Issue by aj-bartocci** [DataDog/dd-sdk-ios#664]
### The issue

When using SPM and linking with the 'Datadog' framework, the app crashes on real devices. The app will run in simulator when the debugger is attached, however if the app is killed in simulator and then manually opened without the debugger it will crash. 

Error: 
```
Library not loaded: @rpath/Datadog.framework/Datadog
Reason: tried: '/usr/lib/swift/Datadog.framework/Datadog' (no such file)
```

When linking against the 'DatadogStatic' framework the app will not crash. Should we be using the 'DatadogStatic' framework when building for production devices? I don't see anything in the docs about this, so just want to make sure we are doing the right thing. 

---

#### Datadog SDK version:

1.17.2

#### Last working Datadog SDK version:

Unsure

#### Dependency Manager:

SPM

#### Other toolset:

Cocoapods

#### Xcode version:

Xcode 13.0

#### Swift version:

Swift 5

#### Deployment Target:

iOS 11+

#### macOS version:

macOS BigSur 11.4 (20F71) (M1 Mac)

**Comment by buranmert** [comment]
hi @aj-bartocci 👋 

your issue might be similar to this one: https://github.com/DataDog/dd-sdk-ios/issues/624#issuecomment-950959458
can you please make sure that you are `Embed & Sign`ing `Datadog` target?

we provide `Datadog` (dynamic) and `DatadogStatic` targets in our package since SPM doesn't always make the right decision about linking type: for example if you want to use `Datadog` in your app extension you'll need dynamic linking whereas in your framework target you may need static linking. 

tl;dr: you can use either `Datadog` or `DatadogStatic` 👌

**Comment by aj-bartocci** [comment]
Awesome thank you for the info, we will use the `DatadogStatic` framework as we don't have an app extension.
